src/Schema/Schema.php

```diff
@@ -437,10 +437,7 @@ public function toDropSql(AbstractPlatform $platform)
      */
     public function getMigrateToSql(Schema $toSchema, AbstractPlatform $platform)
     {
-        $comparator = new Comparator();
-        $schemaDiff = $comparator->compare($this, $toSchema);
-
-        return $schemaDiff->toSql($platform);
+        return Comparator::compareSchemas($this, $toSchema)->toSql($platform);
```

Comment:
This is the only static `Comparator` method, so I'd rather replace it with a non-static one for consistency with the rest of the API and deprecated the `compare()` method since its purpose is unclear from the name.

Please also see https://github.com/doctrine/dbal/pull/4659#discussion_r642103628.

Reply:
Looking at the code and how it is used, it is quite odd the other static diffing methods are not present. Specially given the `Comparator` object does not have any state.

As mentionned in https://github.com/doctrine/dbal/pull/4659#discussion_r642103628, if the goal is to rename all methods anyways, maybe we could introduce a new `Differ` class with the following public API:

```php
class Differ
{
    public static function compareSchemas(Schema $s1, Schema $s2): SchemaDiff;
    public static function compareSequence(Sequence $s1, Sequence $s2): bool;
    public static function compareTable(Table $t1, Table $t2): TableDiff|false;
    public static function compareForeignKey(ForeignKeyConstraint $k1, ForeignKeyConstraint $k2): bool;
    public static function compareColumn(Column $c1, Column $c2): string[]; // could also be ColumnDiff
    public static function compareIndex(Index $i1, Index $i2): bool;
}
```

(Now that I see the API, maybe the methods returning a boolean should be renamed for clarity or moved to a new class.) 

What do you think ?

Reply:
"Differ" is a bit odd name, I'd prefer the same class name: the comparator compares things and returns a diff – looks like a solid naming foundation to me. As for the specific method names, I think they should contain the plural form of the object (e.g. `compareColumns()`).

IIRC, we'll need to only migrate `compareSchemas()` from being static to non-static which doesn't introduction of a new class. We can deprecate non-static calls to the method (which PHP allows) and make it non-static in the next major release.

Reply:
"Differ" sounded great from my suburb of Paris, France... :wink: 

I did not think about evolving the `Comparator` API further, you are quite right!

Reply:
> "Differ" sounded great from my suburb of Paris, France... 😉

I think that would be perfect in the Java world where they smash an _-er_ on top of any noun :-)